src/lib/js/components/fields/field.js

```diff
@@ -310,7 +312,10 @@ export default class Field extends Component {
     const prevData = clone(this.data)
     const { action = {} } = controls.get(prevData.config.controlId)
     prevData.id = `prev-${this.id}`
-    prevData.action = action
+    prevData.action = Object.entries(action).reduce((acc, [key, value]) => {
+      acc[key] = value.bind(this)
```

Comment:
not sure about this one but it binds the Field instance so we have access to `setData` from the component class.